ArmRobot/Assets/Scripts/Robot/RobotVisionCapture.cs

```diff
@@ -0,0 +1,111 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+
+public class RobotVisionCapture : MonoBehaviour
+{
+    public GameObject cube;
+    public GameObject cylinder;
+    public GameObject robot;
+   
+    public int nbMaxGameobjectScene = 10;
+    public float yAltitudeTable = 0.778f;
+    public float minimumDistanceWithoutObjects = 0.05f;
+    public float scaleObject = 0.1f;
+
+    
+    public float robotMinReach = 0.2f;
+    public float robotMaxReach = 0.5f;
+
+
+    void Start()
+    {
+        // first we set the position of the cube 
```

Comment:
Most of this could be moved out into a `CreateObjects` method, which would make this class easier to read.